### PR TITLE
Acceptance spec for adding a billing address

### PR DESCRIPTION
**Pull request by craigjbass**

### CODE REVIEW

File: src/main/kotlin/uk/co/craigbass/pratura/boundary/AddShippingAddress.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.madetech.clean.usecase.SynchronousUseCase
+import uk.co.craigbass.pratura.boundary.AddShippingAddress.Request
+
+interface AddShippingAddress : SynchronousUseCase<Request, Unit> {
+  data class Request(
+    val name: String,
+    val companyName: String?,
+    val addressLine1: String,
+    val addressLine2: String?,
```
Comment (craigjbass) [inline src/main/kotlin/uk/co/craigbass/pratura/boundary/AddShippingAddress.kt]: I'm wondering where this should be `addressLines: List<String>` so addresses can have arbitrary amounts of lines... but then we'd need to enforce validation rules to occur e.g. you must have at least one line

Comment (tomverran) [inline src/main/kotlin/uk/co/craigbass/pratura/boundary/AddShippingAddress.kt]: A NonEmptyList type probably exists either in Kotlin or is trivial to make

But as nice as the multi address line model is it'll probably give you headaches interoperating with third parties that use the standard line1/line2 deal

Comment (craigjbass) [inline src/main/kotlin/uk/co/craigbass/pratura/boundary/AddShippingAddress.kt]: Yeah perhaps. Although I just see it as addressLine1 becoming addressLines[0]

Comment (tomverran) [inline src/main/kotlin/uk/co/craigbass/pratura/boundary/AddShippingAddress.kt]: Yeah in theory though people would probably put in stuff like

Blue door  <-- specific delivery instructions
The Conifers <-- Fuzzy house name
123 Fake St <-- actual line 1

And then say with Stripe you'd accidentally end up chopping off the actual address line 1 and having card declines

I can see the appeal of more flexibility though so hmm

Comment (craigjbass) [inline src/main/kotlin/uk/co/craigbass/pratura/boundary/AddShippingAddress.kt]: I suspect the UI would still give the user address line 1, for UK... but think this localisation thing of presenting the user with 1 or 5 rows should be up to whatever UI is slapped on top

**Pull request merged**